### PR TITLE
cmake: update to 3.21.2

### PR DESCRIPTION
**Pull request by FnControlOption**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by FnControlOption** [comment]
`patch-restore-no-default-rpath-added-on-Tiger.diff` is not needed anymore because of [cmake commit 1af23c4d](https://gitlab.kitware.com/cmake/cmake/-/commit/1af23c4de2c7d58c06171d70b37794b5c860d5f4)

File: devel/cmake/Portfile [hunk 1]
```diff
@@ -36,18 +36,18 @@ subport cmake-devel {
     PortGroup       obsolete 1.0
     replaced_by     cmake
     version         20200323-8638c49f
-    revision        2
+    revision        0
```
Comment (michaelld) [inline devel/cmake/Portfile:39]: undo this ... this section is not to be touched except for removal entirely (which is coming)

**Review by michaelld — CHANGES_REQUESTED** [review]
I've been using the 3.21 series since it was released with no issues, so I have no issues with this PR except the tweak to the `devel` subport. Also: please clean up the remaining patches so that they apply cleanly.

**Comment by michaelld** [comment]
Tagging @kencu ... maybe you have thoughts on how/if this new CMake will work better or worse on older macOS? I know from discussion upstream that they accepted the Tiger fix, and in that discussion was noted that there's no reason CMake won't work on MacOS X back to 10.4. But, I don't think they do CI / QA testing beyond 10.12... hence the question here before this gets merged.

**Comment by kencu** [comment]
Thanks Michael. I'll build this today and report back when done.

**Review by kencu — APPROVED** [review]

**Comment by kencu** [comment]
builds without issue on 10.6. 

If something is broken before that, we can straighten it out later on.

Thanks for asking.

For the future, if cmake has tests available (must have ... ), perhaps we might figure out how to enable and run the test suite.

**Review by michaelld — APPROVED** [review]
LGTM; thx!

**Comment by michaelld** [comment]
> builds without issue on 10.6.
> 
> If something is broken before that, we can straighten it out later on.
> 
> Thanks for asking.
> 
> For the future, if cmake has tests available (must have ... ), perhaps we might figure out how to enable and run the test suite.

Excellent all around! Yes cmake does have built-in tests. In my handling of the cmake port, I have never seen this pass at 100% ; there are almost always 10-15 tests that fail for various reasons -- out of something like 350, so that's pretty decent. Some I could fix easily, some would take serious work. I have, obviously, never taken time for either & right now I really don't have time. That said, it should be straight forward to add a `test` phase to this port ... hmmm ... have to look into that.

Anyway: Merging!

**Pull request merged**